main/mySpace/all_attendance_list.php

```diff
@@ -0,0 +1,1185 @@
+<?php
+
+/* For licensing terms, see /license.txt */
+
+use Chamilo\CourseBundle\Entity\CLpCategory;
+use ChamiloSession as Session;
+
+if (!isset($_GET['course'])) {
+    $cidReset = true;
+}
+
+require_once __DIR__.'/../inc/global.inc.php';
+
+api_block_anonymous_users();
+
+$htmlHeadXtra[] = '<script type="text/javascript" src="'.api_get_path(WEB_PUBLIC_PATH).'assets/jquery.easy-pie-chart/dist/jquery.easypiechart.js"></script>';
+
+$export = isset($_GET['export']) ? $_GET['export'] : false;
+$sessionId = isset($_GET['id_session']) ? (int) $_GET['id_session'] : 0;
+$origin = api_get_origin();
+$course_code = isset($_GET['course']) ? Security::remove_XSS($_GET['course']) : '';
+$courseInfo = api_get_course_info($course_code);
+$courseCode = '';
+if ($courseInfo) {
+    $courseCode = $courseInfo['code'];
+}
+$student_id = isset($_GET['student_id']) ? (int) $_GET['student_id'] : 0;
+$coachId = isset($_GET['id_coach']) ? (int) $_GET['id_coach'] : 0;
+$details = isset($_GET['details']) ? Security::remove_XSS($_GET['details']) : '';
+$currentUrl = api_get_self().'?student='.$student_id.'&course='.$courseCode.'&id_session='.$sessionId
+    .'&origin='.$origin.'&details='.$details;
+$allowMessages = api_get_configuration_value('private_messages_about_user');
+
+
+
+
+ // Varible for all attendance list
+
+
+$startDate = new DateTime();
+$startDate = $startDate->modify('-1 week');
+if(isset($_GET['startDate'])){
+    $startDate = new DateTime($_GET['startDate']);
+
+}
+$startDate = $startDate->setTime(0,0,0);
+
+$endDate =  new DateTime();
+
+if(isset($_GET['endDate'])){
+    $endDate = new DateTime($_GET['endDate']);
+
+}
+$endDate = $endDate->setTime(23,59,0);
+// $startDate = new DateTime(api_get_local_time($startDate));
+// $endDate = new DateTime(api_get_local_time($endDate));
+if($startDate > $endDate){
+    $a = $startDate;
```

Comment:
Remove variables that are not being used $a? $b?

Reply:
yes, sure, is temp data. Already change to datetemp